openapi/openapi/src/main/scala/endpoints4s/openapi/Urls.scala

```diff
@@ -83,6 +83,32 @@ trait Urls extends algebra.Urls {
   )
   type QueryStringParam[A] = DocumentedQueryStringParam[A]
 
+  //TODO: This is kinda ugly. But discriminated schemas make little sense on query strings?
+  //Stil maybe this should be done better, but I haven't yet fully understood the openapi schema to figure out how to deal with nesting here
+  private def unpackSchema[A](qsp:QueryStringParam[A]):List[Schema] = qsp.schema match {
```

Comment:
Indeed, this is a signal that tells us that something is wrong with our approach. Most likely we need an intermediate structure somewhere to avoid this.